apps/electron/src/helper/dialog/dialog.ts

```diff
@@ -217,6 +219,40 @@ export async function loadDBFile(): Promise<LoadDBFileResult> {
       mainDBPath: internalFilePath,
     });
 
+    const db = await ensureSQLiteDB(workspaceId);
+    const { Workspace, Schema } = await import('@blocksuite/store');
+    const schema = new Schema();
+    const { __unstableSchemas, AffineSchemas } = await import(
+      '@blocksuite/blocks/models'
+    );
+    schema.register(AffineSchemas).register(__unstableSchemas);
+    const workspace = new Workspace({
+      id: workspaceId,
+      schema,
+      blobStorages: [],
+    });
+    const downloadBinary = async (doc: Doc): Promise<void> => {
+      const update = db.getDocAsUpdates(doc.guid);
+      if (update) {
+        applyUpdate(doc, update);
+      }
+      await Promise.all(
+        [...doc.subdocs].map(subdoc => {
+          return downloadBinary(subdoc);
+        })
+      );
+    };
+    await downloadBinary(workspace.doc);
+    // always try to upgrade the workspace
+    schema.upgradeWorkspace(workspace.doc);
+    const uploadBinary = (doc: Doc) => {
+      db.applyUpdate(encodeStateAsUpdate(doc), 'self', doc.guid);
+      [...doc.subdocs].map(subdoc => {
+        return uploadBinary(subdoc);
+      });
+    };
+    uploadBinary(workspace.doc);
+
```

Comment:
can you please merge the logic with `migrateToSubdocAndReplaceDatabase`, which clones a copy of the input db and then write back to the embedded DB file directly?